src/common/include/gudhi/random_graph_generators.h

```diff
@@ -0,0 +1,98 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Vincent Rouvreau
+ *
+ *    Copyright (C) 2024 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef RANDOM_GRAPH_GENERATORS_H_
+#define RANDOM_GRAPH_GENERATORS_H_
+
+#include <gudhi/Random.h>
+
+
+#include <algorithm>  // for std::prev_permutation
+#include <vector>
+#include <array>
+#include <random>
+#include <numeric>  // for std::iota
+#include <cstddef>  // for std::size_t
+#include <type_traits>  // for std::is_floating_point_v
+#ifdef DEBUG_TRACES
+#include <iostream>
+#endif  // DEBUG_TRACES
+
+std::random_device rd;
+
+namespace Gudhi {
+
+template <typename Vertex_handle>
+std::vector<std::array<Vertex_handle, 2>> random_edges(Vertex_handle nb_vertices, double density = 0.15) {
```

Comment:
The algorithm in there seems very complicated. What do you want exactly?
1. For each pair of vertices, add the edge with proba 15% (independently)
2. Get exactly 15% * n(n-1)/2 edges, chosen randomly
3. Other